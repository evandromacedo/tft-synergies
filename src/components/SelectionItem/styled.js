import styled from 'styled-components';
import SynergyIconComponent from '../SynergyIcon';

export const Wrapper = styled.div`
  max-width: 22.4rem;
  display: grid;
  grid-template-columns: 6.2rem 1fr;
  align-items: center;
  grid-gap: 0.6rem;
  border: 1px dashed ${props => props.theme.background4};
  cursor: move;
`;

export const IconAndDescription = styled.div`
  display: inline-grid;
  grid-auto-flow: column;
  grid-gap: 0.6rem;
  align-items: center;
  justify-content: start;
  padding-right: 0.8rem;
`;

export const Img = styled.img`
  width: 6.2rem;
`;

export const SynergyIcon = styled(SynergyIconComponent)`
  font-size: 3.2rem;
`;

export const Description = styled.p`
  ${props => props.theme.sans14pxItalic}
  line-height: 1.8rem;
`;
