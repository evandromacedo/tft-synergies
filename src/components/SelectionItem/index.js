import React from 'react';
import PropTypes from 'prop-types';
import * as S from './styled';

export default function SelectionItem({ item }) {
  function getImgSrc(name) {
    return `https://rerollcdn.com/items/${name.replace(/ /g, '')}.png`;
  }

  const { key, name, bonus } = item;

  const synergyIcons = {
    youmuusghostblade: 'assassin',
    knightsvow: 'knight',
    fronzenmallet: 'glacial',
    yuumi: 'sorcerer',
    bladeoftheruinedking: 'blademaster',
    darkin: 'demon'
  };

  return (
    <S.Wrapper>
      <S.Img src={getImgSrc(name)} alt={name} title={name} />
      <S.IconAndDescription>
        <S.SynergyIcon classOrOrigin={synergyIcons[key]} type="default" />
        <S.Description>{bonus}</S.Description>
      </S.IconAndDescription>
    </S.Wrapper>
  );
}

SelectionItem.propTypes = {
  item: PropTypes.object
};
