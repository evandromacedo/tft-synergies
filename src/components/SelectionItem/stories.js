import React from 'react';
import SelectionItem from '.';
import { storiesOf } from '@storybook/react';

const item = {
  key: 'bladeoftheruinedking',
  name: 'Blade of the Ruined King',
  type: 'Offensive',
  bonus: 'Wearer is also a Blademaster.',
  tier: 1,
  depth: 2,
  buildsFrom: ['spatula', 'recurvebow'],
  kind: 'advanced',
  champs: ['Ashe', 'Yasuo', 'Zed']
};

storiesOf('Selection Item', module).add('Default', () => <SelectionItem item={item} />);
