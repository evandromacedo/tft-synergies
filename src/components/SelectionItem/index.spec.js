import React from 'react';
import { shallow } from 'enzyme';
import { wrapInTestContext } from 'react-dnd-test-utils';
import toJSON from 'enzyme-to-json';
import SelectionItem from '.';

const botrkMock = {
  key: 'bladeoftheruinedking',
  name: 'Blade of the Ruined King',
  type: 'Offensive',
  bonus: 'Wearer is also a Blademaster.',
  tier: 1,
  depth: 2,
  buildsFrom: ['spatula', 'recurvebow'],
  kind: 'advanced',
  champs: ['Ashe', 'Yasuo', 'Zed']
};

const SelectionItemWrapped = wrapInTestContext(SelectionItem);

describe('<SelectionItem />', () => {
  it('renders properly', () => {
    shallow(<SelectionItemWrapped item={botrkMock} />);
  });

  it('renders and matches snapshot', () => {
    const wrapper = shallow(<SelectionItemWrapped item={botrkMock} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
