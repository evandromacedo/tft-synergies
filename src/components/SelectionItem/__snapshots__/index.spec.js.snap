// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectionItem /> renders and matches snapshot 1`] = `
<Memo()
  backend={[Function]}
>
  <SelectionItem
    item={
      Object {
        "bonus": "Wearer is also a Blademaster.",
        "buildsFrom": Array [
          "spatula",
          "recurvebow",
        ],
        "champs": Array [
          "Ashe",
          "Yasuo",
          "Zed",
        ],
        "depth": 2,
        "key": "bladeoftheruinedking",
        "kind": "advanced",
        "name": "Blade of the Ruined King",
        "tier": 1,
        "type": "Offensive",
      }
    }
  />
</Memo()>
`;
